### ISSUE TITLE
ignoreSourceCodeByRegex option is ignored between // @codeCoverageIgnoreStart and // @codeCoverageIgnoreEnd

### ISSUE THREAD

**Issue by VincentLanglet** [infection/infection#1561]
I have multiple mutation
```
33) /Users/vincentl/Perso/Twig-CS-Fixer/src/Token/Tokenizer.php:476    [M] Throw_

--- Original
+++ New
@@ @@
             $this->pushToken(Token::DQ_STRING_END_TYPE, $match[0], $bracket);
             $this->moveCursor($match[0]);
         } else {
-            // @codeCoverageIgnoreStart
-            throw new LogicException(sprintf('Unhandled character "%s" in lexDqString.', $this->code[$this->cursor]));
+            new LogicException(sprintf('Unhandled character "%s" in lexDqString.', $this->code[$this->cursor]));
             // @codeCoverageIgnoreEnd
         }
     }
```
Which I don't succeed to ignore

I tried
```
"mutators": {
        "@default": true,
        "global-ignoreSourceCodeByRegex": [
            "assert\\(.*",
            "throw new LogicException\\(.*"
        ]
    },
```
with no success,
I also tried to ignore
```
"\\/\\/ @codeCoverageIgnoreStart.*"
```
instead but nothing changed.

And I clearly don't understand what is wrong with my regex

**Comment by sanmai** [comment]
Try this:
```php
// @infection-ignore-all 
throw new LogicException()
```

Compare these two:
- https://infection-php.dev/r/2rkr
- https://infection-php.dev/r/0yp2

**Comment by VincentLanglet** [comment]
> Try this:
> 
> ```
> // @infection-ignore-all 
> throw new LogicException()
> ```

Sure, but since since I throw multiple logic exception, I would prefer to use a global regex rule.

Compare 
- https://infection-php.dev/r/5gnp
- https://infection-php.dev/r/4kwo

Adding a comment is breaking the ignore rule

**Comment by VincentLanglet** [comment]
I think it's a bug: https://infection-php.dev/r/l0r2

**Comment by VincentLanglet** [comment]
And it's seems related to `// @codeCoverageIgnoreStart`

https://infection-php.dev/r/r5w6
https://infection-php.dev/r/0yx2
https://infection-php.dev/r/4kdo

**Comment by sanmai** [comment]
So it looks like `ignoreSourceCodeByRegex` is not respected for uncovered mutations.

**Comment by VincentLanglet** [comment]
> So it looks like `ignoreSourceCodeByRegex` is not respected for uncovered mutations.

Any idea how to fix this/where I should look at to fix this ?
This bug is kinda annoying for the reporting

**Comment by maks-rafalko** [comment]
Thanks for reporting this issue, fixed and released as `0.25.3`

Before: https://infection-php.dev/r/4kdo
After: https://infection-php.dev/r/vr21
